File: teste-aiko/src/types/EquipmentStateHistory.ts
import { EquipmentStateJson } from "./EquipmentState"

export type EquipmentStateHistoryJson = {
  equipmentId: string,
  states: Array<EquipmentStateHistoryItemJson>
}

export type EquipmentStateHistoryItemJson = {
  date: string,
  equipmentStateId: string
}

export type EquipmentStateHistoryItem = EquipmentStateHistoryItemJson & EquipmentStateJson 

File: aiko/Models/EquipmentStateHistory.cs
using System;
using System.Collections.Generic;

namespace aiko.Models
{
    public class EquipmentStateHistory
    {
        public string EquipmentId { get; set; }
        public List<StateHistory> States { get; set; }
    }

    public class StateHistory
    {
        public DateTime Date { get; set; }
        public string EquipmentStateId { get; set; }
    }
}
